src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java

```diff
@@ -163,39 +162,38 @@ private void validateValueType(FailureCollector collector) {
 
   private void validateDateRange(FailureCollector collector) {
     if (containsMacro(ServiceNowConstants.PROPERTY_START_DATE) ||
-      containsMacro(ServiceNowConstants.PROPERTY_END_DATE)) {
+      containsMacro(ServiceNowConstants.PROPERTY_END_DATE) ||
+      (Util.isNullOrEmpty(startDate) && Util.isNullOrEmpty(endDate))) {
       return;
     }
 
-    if (!Util.isNullOrEmpty(startDate) && !Util.isNullOrEmpty(endDate) &&
-      !Util.isValidDateFormat(ServiceNowConstants.DATE_FORMAT, startDate) &&
-      !Util.isValidDateFormat(ServiceNowConstants.DATE_FORMAT, endDate)) {
-      collector.addFailure("Invalid format for Start date. Correct Format: " +
-          ServiceNowConstants.DATE_FORMAT, null)
-        .withConfigProperty(ServiceNowConstants.PROPERTY_START_DATE);
-      collector.addFailure("Invalid format for End date. Correct Format:" +
-          ServiceNowConstants.DATE_FORMAT, null)
-        .withConfigProperty(ServiceNowConstants.PROPERTY_END_DATE);
-      return;
-    }
     // validate the date formats for both start date & end date
-    if (!Util.isNullOrEmpty(startDate) && !Util.isValidDateFormat(ServiceNowConstants.DATE_FORMAT, startDate)) {
-      collector.addFailure("Invalid format for Start date. Correct Format: " +
-          ServiceNowConstants.DATE_FORMAT, null)
-        .withConfigProperty(ServiceNowConstants.PROPERTY_START_DATE);
-      return;
+    if (!Util.isNullOrEmpty(startDate)) {
+      if (Util.isNullOrEmpty(endDate)) {
+        collector.addFailure("An end date must be given when a start date is given.",
+                             "Enter an end date, or remove the start date")
+          .withConfigProperty(ServiceNowConstants.PROPERTY_END_DATE);
+      }
+      if (!Util.isValidDateFormat(ServiceNowConstants.DATE_FORMAT, startDate)) {
+        collector.addFailure("Invalid format for Start date. Correct Format: " +
+                               ServiceNowConstants.DATE_FORMAT, null)
+          .withConfigProperty(ServiceNowConstants.PROPERTY_START_DATE);
+      }
     }
 
-    if (!Util.isNullOrEmpty(endDate) && !Util.isValidDateFormat(ServiceNowConstants.DATE_FORMAT, endDate)) {
-      collector.addFailure("Invalid format for End date. Correct Format:" +
-          ServiceNowConstants.DATE_FORMAT, null)
-        .withConfigProperty(ServiceNowConstants.PROPERTY_END_DATE);
-      return;
-    }
-
-    if (Util.isNullOrEmpty(startDate) || Util.isNullOrEmpty(endDate)) {
-      return;
+    if (!Util.isNullOrEmpty(endDate)) {
+      if (Util.isNullOrEmpty(startDate)) {
+        collector.addFailure("A start date must be given when a end date is given.",
+                             "Enter an start date, or remove the end date")
```

Comment:
nit: "Enter **a** start date" as 'start' starts with a consonant. Similarly in the line above, "when **an** end date is given".

Reply:
Fixed. 70325997c4d19a7462f18ae0264b4de41a156f8e